lib/app.jsx

```diff
@@ -84,6 +87,28 @@ const isElectron = ( () => {
 	return () => foundElectron;
 } )();
 
+const when = ( condition ) => ( action, fallback = noop ) => ( ... args ) => (
+	condition( ... args ) ? action( ... args ) : fallback( ... args )
+)
+
+const preventDefaultThen = ( fn ) => ( e ) => {
+	e.preventDefault()
+	return fn( e )
+}
+
+const whenCanResize = when( ( { canResize } ) => canResize );
+const whenIsResizing = when( ( { isResizing } ) => isResizing );
+
+const detectCanResize = ( { onResizable, sourceListSelector } ) => ( e ) => {
+	const node = document.querySelector( sourceListSelector )
+	if ( !node ) return;
+	onResizable( e.clientX - Math.abs( node.getBoundingClientRect().right ) < 3 )
+}
+
+const makeResizable = whenIsResizing( ( { onUpdateWidth } ) => ( e ) => onUpdateWidth( e.clientX ), detectCanResize )
+const startResizing = whenCanResize( ( { onStartResizing } ) => preventDefaultThen( () => onStartResizing() ) )
+const stopResizing = whenIsResizing( ( { onStopResizing } ) => preventDefaultThen( onStopResizing ) )
```

Comment:
I feel like we should just include [Ramda](http://ramdajs.com/0.20.1/docs/) in this project since its helpers would be awesome in places like this. I find these sequences difficult to read because although they feel like a chain or sequence of functions, they are actually passing a bunch of functions through the parameters and I've had to go over it a few times before realizing what exactly they are doing.

```js
import {
	compose,
	ifElse,
	invoker,
	prop,
	tap,
	when
} from 'ramda';

const preventDefault = tap( invoker( 'preventDefault' ) );

const makeResizable = ifElse(
	prop( 'isResizing' ), 
		( { onStartResizing } ) => compose( onStartResizing, prop( 'clientX' ) ),
		detectCanResize
);

const startResizing = when( prop( 'canResize' ), compose( prop( 'onStartResizing' ), preventDefault ) );

const stopResizing = when( prop( 'isResizing' ), compose( prop( 'onStopResizing' ), preventDefault ) );
```

That's an idea using Ramda. I like what you are going for with the functional style, but I think it would be simpler/clearer without the middle step.